src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -558,10 +557,16 @@ private void openapi(RoutingContext routingContext) {
     }
 
     private void metrics(RoutingContext routingContext) {
-        routingContext.response()
+        if (metricsCollector != null) {
+            routingContext.response()
                 .putHeader("Content-Type", "text/plain; version=0.0.4; charset=utf-8")
                 .setStatusCode(HttpResponseStatus.OK.code())
-                .end(metricsReporter.scrape());
+                .end(metricsCollector.scrape());
+        } else {
+            HttpBridgeError error = new HttpBridgeError(HttpResponseStatus.NOT_FOUND.code(), "The metrics endpoint is not enabled.");
+            HttpUtils.sendResponse(routingContext, HttpResponseStatus.NOT_FOUND.code(),
+                BridgeContentType.KAFKA_JSON, JsonUtils.jsonToBytes(error.toJson()));
```

Comment:
What exactly is the JSON we return here? Is it compatibile with the metrics/monitoring tools? Would HTTP 404 without any message be more sensible?

Reply:
That's a good point. I asked @fvaleri to do the same way as we do for other endpoints but in this case returning a JSON doesn't make sense because expected content for metrics endpoint is just text/plain. So I agree just 404 with empty body is better (as first proposal by Fede).